solidity/dashboard/src/sagas/liquidity-rewards.js

```diff
@@ -60,6 +64,11 @@ function* fetchLiquidityRewardsData(liquidityRewardPair, address) {
       )
       // % of total pool in the `LPRewards` contract.
       shareOfPoolInPercent = percentageOf(lpBalance, totalSupply).toString()
+
+      lpTokenBalance = yield call(
+        [LiquidityRewards, LiquidityRewards.calculateLPTokenBalance],
+        shareOfPoolInPercent
```

Comment:
I think we can't calculate LP token balance correctly based on the % of total pool in the `LPRewards` contract. I think we should:
* get the deposited LP tokens in `LPRewards` contract for a given user,
* calculate % of **Uniswap** pool based on an amount above,
* and calculate token balance based on this formula https://uniswap.org/docs/v2/advanced-topics/understanding-returns/#why-is-my-liquidity-worth-less-than-i-put-in

Thoughts? 🤔

Reply:
@Battenfield We wonder if our understanding is correct. Could you please take a look at it?